src/parser/asts/functions.c

```
@@ -1,215 +1,168 @@
+    AST_FUNCTION_DEC* func = malloc(sizeof(AST_FUNCTION_DEC));
+    func->returnType = malloc(1);
+
+    func->type = AST_TYPE_FUNCTION_DECLARATION;
+
+    int offset;
+
+    switch(result.tokens[index + 2].type) {
+        case PAREN_OPEN:
+            if(result.tokens[index + 1].type != KEYWORD) {
+                printf("Error: Excepted a keyword as function name!\n");
+                return NULL;
```

cppcheck (2.10-2)

[error] Memory leak

(memleak)